File: swift-aws/Sources/env.swift
import SwiftDotenv

enum EnvError: Error {
	case noSuchEnvironmentVariable(String)
}

func assertEnvVar(_ name: String) throws -> String {
	if let result = Dotenv[name]?.stringValue {
		return result
	} else {
		throw EnvError.noSuchEnvironmentVariable(name)
	}
}

File: ts-openai/src/env.ts
import dotenv from "dotenv";

dotenv.config();

export function assertEnvVar(name: string): string {
	const result = process.env[name];
	if (!result) {
		throw new Error(`expected environment variable: ${name}`);
	}
	return result;
}

export function assertIntEnvVar(name: string): number {
	const s = assertEnvVar(name);
	if (!/^[0-9]+$/.exec(s)) {
		throw new Error(`environment variable doesn't look like an integer: ${name}`);
	}
	return parseInt(s, 10);
}